### PR TITLE
fix: fixed the duplicate acls

### PR DESCRIPTION
**Pull request by Ak-sky**
### Description

Some of the ACLs were getting added twice as it was coming from both pattern and VPC module, to fix this, ACL rules for internal IBM traffic will be added from VPC module only.

fixes https://github.com/terraform-ibm-modules/terraform-ibm-landing-zone/issues/393

#### No release required

- [ ] Examples or tests (addition or updates of examples or tests)
- [ ] Documentation update
- [ ] CI-related update (pipeline, etc.)
- [ ] Other changes that don't affect Terraform code

#### Release required

- [x] Bug fix (patch release (`x.x.X`): Change that fixes an issue and is compatible with earlier versions)
- [ ] New feature (minor release (`x.X.x`): Change that adds functionality and is compatible with earlier versions)
- [ ] Breaking change (major release (`X.x.x`): Change that is likely incompatible with previous versions)

##### Release notes content

The [terraform-ibm-landing-zone-vpc](https://github.com/terraform-ibm-modules/terraform-ibm-landing-zone-vpc) module used by this solution will now add appropriate IBM Cloud ACL rules for all created VPCs. This solution was also adding those same rules. The result was VPC ACLs that had duplicate rules.

This release will remove the IBM Cloud ACL rules from this solution, and allow the terraform-ibm-landing-zone-vpc module to add these rules when creating the VPC.

NOTE: when upgrading to this release, you may notice that existing ACLs created by previous versions will be marked as Terraform as needing to be removed or updated. This is expected, as the duplicate rules would be removed from existing ACLs.

---

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- Merge by using "Squash and merge".
- Use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author.

    The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).

### CODE REVIEW

**Comment by ocofaigh** [comment]
@Ak-sky Can you fill in the release notes section please? Ensure to include what the consumers experience will be (aka what happens to the current deployed ACLs)

**Review by vburckhardt — CHANGES_REQUESTED** [review]
Thanks for the PR 👍  What do you think about cleaning up / removing the default_vpc_rules at https://github.com/terraform-ibm-modules/terraform-ibm-landing-zone/blob/4dc40bd5abe011b2b55581d23b67123ff672ba5d/patterns/dynamic_values/config_modules/acl_rules/acl_rules.tf#L5 if this is not longer in use?

**Comment by vburckhardt** [comment]
Could you also show the network acls after applying this new version?

**Comment by Ak-sky** [comment]
> Could you also show the network acls after applying this new version?

For both management and workload ACLs (here refering only management VPC) -
`module.landing_zone.module.vpc["management"].ibm_is_network_acl.network_acl["management-acl"]`

Before - 

> {
    "rules": [
        {
            "action": "allow",
            "destination": "10.0.0.0/8",
            "direction": "inbound",
            "icmp": [],
            "id": "1f422448-3617-4952-aecd-6247d8cb7e62",
            "ip_version": "ipv4",
            "name": "ibmflow-iaas-inbound",
            "source": "161.26.0.0/16",
            "subnets": 7,
            "tcp": [],
            "udp": []
        },
        {
            "action": "allow",
            "destination": "161.26.0.0/16",
            "direction": "outbound",
            "icmp": [],
            "id": "9eff766d-d42f-4f2c-931e-be770be327ee",
            "ip_version": "ipv4",
            "name": "ibmflow-iaas-outbound",
            "source": "10.0.0.0/8",
            "subnets": 7,
            "tcp": [],
            "udp": []
        },
        {
            "action": "allow",
            "destination": "10.0.0.0/8",
            "direction": "inbound",
            "icmp": [],
            "id": "8d7b2d0b-81b4-43dc-bf1e-063606bcf33e",
            "ip_version": "ipv4",
            "name": "ibmflow-paas-inbound",
            "source": "166.8.0.0/14",
            "subnets": 7,
            "tcp": [],
            "udp": []
        },
        {
            "action": "allow",
            "destination": "166.8.0.0/14",
            "direction": "outbound",
            "icmp": [],
            "id": "a736b28a-9c7a-49fc-839e-d8ee036656f6",
            "ip_version": "ipv4",
            "name": "ibmflow-paas-outbound",
            "source": "10.0.0.0/8",
            "subnets": 7,
            "tcp": [],
            "udp": []
        },
        {
            "action": "allow",
            "destination": "10.0.0.0/8",
            "direction": "inbound",
            "icmp": [],
            "id": "ab2aac7f-618e-4b8d-ae49-ddb35ce2cfec",
            "ip_version": "ipv4",
            "name": "ibmflow-allow-vpc-connectivity-inbound-3e78",
            "source": "10.10.20.0/24",
            "subnets": 7,
            "tcp": [],
            "udp": []
        },
        {
            "action": "allow",
            "destination": "10.0.0.0/8",
            "direction": "inbound",
            "icmp": [],
            "id": "223b0b6d-2f4b-409f-b129-d038c6fa62a5",
            "ip_version": "ipv4",
            "name": "ibmflow-allow-vpc-connectivity-inbound-33aa",
            "source": "10.20.20.0/24",
            "subnets": 7,
            "tcp": [],
            "udp": []
        },
        {
            "action": "allow",
            "destination": "10.0.0.0/8",
            "direction": "inbound",
            "icmp": [],
            "id": "24e123b3-f691-49de-a28b-e97bd430f6fb",
            "ip_version": "ipv4",
            "name": "ibmflow-allow-vpc-connectivity-inbound-13be",
            "source": "10.10.30.0/24",
            "subnets": 7,
            "tcp": [],
            "udp": []
        },
        {
            "action": "allow",
            "destination": "10.0.0.0/8",
            "direction": "inbound",
            "icmp": [],
            "id": "0bf752f3-77bd-4e47-86b0-bbc687c3aaf3",
            "ip_version": "ipv4",
            "name": "ibmflow-allow-vpc-connectivity-inbound-6c66",
            "source": "10.30.10.0/24",
            "subnets": 7,
            "tcp": [],
            "udp": []
        },
        {
            "action": "allow",
            "destination": "10.0.0.0/8",
            "direction": "inbound",
            "icmp": [],
            "id": "c23eda9e-22e0-4f86-a123-495324a44bdd",
            "ip_version": "ipv4",
            "name": "ibmflow-allow-vpc-connectivity-inbound-db9f",
            "source": "10.30.20.0/24",
            "subnets": 7,
            "tcp": [],
            "udp": []
        },
        {
            "action": "allow",
            "destination": "10.0.0.0/8",
            "direction": "inbound",
            "icmp": [],
            "id": "8b118a02-146f-4045-a77c-ab5bc76c3b11",
            "ip_version": "ipv4",
            "name": "ibmflow-allow-vpc-connectivity-inbound-5cb2",
            "source": "10.20.10.0/24",
            "subnets": 7,
            "tcp": [],
            "udp": []
        },
        {
            "action": "allow",
            "destination": "10.0.0.0/8",
            "direction": "inbound",
            "icmp": [],
            "id": "71e872aa-12e1-45ac-8ba7-ba3f97363299",
            "ip_version": "ipv4",
            "name": "ibmflow-allow-vpc-connectivity-inbound-7fc9",
            "source": "10.10.10.0/24",
            "subnets": 7,
            "tcp": [],
            "udp": []
        },
        {
            "action": "allow",
            "destination": "10.10.20.0/24",
            "direction": "outbound",
            "icmp": [],
            "id": "cf33488c-81c9-4a3f-9781-bbedc6568378",
            "ip_version": "ipv4",
            "name": "ibmflow-allow-vpc-connectivity-outbound-3e78",
            "source": "10.0.0.0/8",
            "subnets": 7,
            "tcp": [],
            "udp": []
        },
        {
            "action": "allow",
            "destination": "10.20.20.0/24",
            "direction": "outbound",
            "icmp": [],
            "id": "70d703bb-4c85-4577-bb01-f0e6efe70ac9",
            "ip_version": "ipv4",
            "name": "ibmflow-allow-vpc-connectivity-outbound-33aa",
            "source": "10.0.0.0/8",
            "subnets": 7,
            "tcp": [],
            "udp": []
        },
        {
            "action": "allow",
            "destination": "10.10.30.0/24",
            "direction": "outbound",
            "icmp": [],
            "id": "9f02131b-4cd3-4ddd-8da7-b09bf5a9881a",
            "ip_version": "ipv4",
            "name": "ibmflow-allow-vpc-connectivity-outbound-13be",
            "source": "10.0.0.0/8",
            "subnets": 7,
            "tcp": [],
            "udp": []
        },
        {
            "action": "allow",
            "destination": "10.30.10.0/24",
            "direction": "outbound",
            "icmp": [],
            "id": "993265b3-7186-4d7c-82f8-c38f263edfd6",
            "ip_version": "ipv4",
            "name": "ibmflow-allow-vpc-connectivity-outbound-6c66",
            "source": "10.0.0.0/8",
            "subnets": 7,
            "tcp": [],
            "udp": []
        },
        {
            "action": "allow",
            "destination": "10.30.20.0/24",
            "direction": "outbound",
            "icmp": [],
            "id": "feddc303-8451-4503-86dd-7c5ece56d3cd",
            "ip_version": "ipv4",
            "name": "ibmflow-allow-vpc-connectivity-outbound-db9f",
            "source": "10.0.0.0/8",
            "subnets": 7,
            "tcp": [],
            "udp": []
        },
        {
            "action": "allow",
            "destination": "10.20.10.0/24",
            "direction": "outbound",
            "icmp": [],
            "id": "8ed0d99c-e430-40d8-8baf-fc88d5570b6e",
            "ip_version": "ipv4",
            "name": "ibmflow-allow-vpc-connectivity-outbound-5cb2",
            "source": "10.0.0.0/8",
            "subnets": 7,
            "tcp": [],
            "udp": []
        },
        {
            "action": "allow",
            "destination": "10.10.10.0/24",
            "direction": "outbound",
            "icmp": [],
            "id": "0909586b-8f2c-4dcc-89d3-ca6b60456c52",
            "ip_version": "ipv4",
            "name": "ibmflow-allow-vpc-connectivity-outbound-7fc9",
            "source": "10.0.0.0/8",
            "subnets": 7,
            "tcp": [],
            "udp": []
        },
        {
            "action": "allow",
            "destination": "10.0.0.0/8",
            "direction": "inbound",
            "icmp": [],
            "id": "eabf7845-906a-44c8-9481-06851dae8805",
            "ip_version": "ipv4",
            "name": "allow-ibm-inbound",
            "source": "161.26.0.0/16",
            "subnets": 7,
            "tcp": [],
            "udp": []
        },
        {
            "action": "allow",
            "destination": "10.0.0.0/8",
            "direction": "inbound",
            "icmp": [],
            "id": "63a3a3fe-d567-4ad2-b6e6-41d432db2a35",
            "ip_version": "ipv4",
            "name": "allow-all-network-inbound",
            "source": "10.0.0.0/8",
            "subnets": 7,
            "tcp": [],
            "udp": []
        },
        {
            "action": "allow",
            "destination": "161.26.0.0/16",
            "direction": "outbound",
            "icmp": [],
            "id": "264aa3d7-a32e-4a1b-97af-7499786edd6c",
            "ip_version": "ipv4",
            "name": "allow-all-outbound",
            "source": "10.0.0.0/8",
            "subnets": 7,
            "tcp": [],
            "udp": []
        },
        {
            "action": "deny",
            "destination": "0.0.0.0/0",
            "direction": "inbound",
            "icmp": [],
            "id": "708fc11a-23f9-46fa-918b-003286c8dc91",
            "ip_version": "ipv4",
            "name": "ibmflow-deny-all-inbound",
            "source": "0.0.0.0/0",
            "subnets": 7,
            "tcp": [],
            "udp": []
        },
        {
            "action": "deny",
            "destination": "0.0.0.0/0",
            "direction": "outbound",
            "icmp": [],
            "id": "a00bf7ba-5883-4e8a-a0e7-b982d32524c1",
            "ip_version": "ipv4",
            "name": "ibmflow-deny-all-outbound",
            "source": "0.0.0.0/0",
            "subnets": 7,
            "tcp": [],
            "udp": []
        }
    ]
}

After -

> {
    "rules": [
        {
            "action": "allow",
            "destination": "10.0.0.0/8",
            "direction": "inbound",
            "icmp": [],
            "id": "1f422448-3617-4952-aecd-6247d8cb7e62",
            "ip_version": "ipv4",
            "name": "ibmflow-iaas-inbound",
            "source": "161.26.0.0/16",
            "subnets": 7,
            "tcp": [],
            "udp": []
        },
        {
            "action": "allow",
            "destination": "161.26.0.0/16",
            "direction": "outbound",
            "icmp": [],
            "id": "9eff766d-d42f-4f2c-931e-be770be327ee",
            "ip_version": "ipv4",
            "name": "ibmflow-iaas-outbound",
            "source": "10.0.0.0/8",
            "subnets": 7,
            "tcp": [],
            "udp": []
        },
        {
            "action": "allow",
            "destination": "10.0.0.0/8",
            "direction": "inbound",
            "icmp": [],
            "id": "8d7b2d0b-81b4-43dc-bf1e-063606bcf33e",
            "ip_version": "ipv4",
            "name": "ibmflow-paas-inbound",
            "source": "166.8.0.0/14",
            "subnets": 7,
            "tcp": [],
            "udp": []
        },
        {
            "action": "allow",
            "destination": "166.8.0.0/14",
            "direction": "outbound",
            "icmp": [],
            "id": "a736b28a-9c7a-49fc-839e-d8ee036656f6",
            "ip_version": "ipv4",
            "name": "ibmflow-paas-outbound",
            "source": "10.0.0.0/8",
            "subnets": 7,
            "tcp": [],
            "udp": []
        },
        {
            "action": "allow",
            "destination": "10.0.0.0/8",
            "direction": "inbound",
            "icmp": [],
            "id": "ab2aac7f-618e-4b8d-ae49-ddb35ce2cfec",
            "ip_version": "ipv4",
            "name": "ibmflow-allow-vpc-connectivity-inbound-3e78",
            "source": "10.10.20.0/24",
            "subnets": 7,
            "tcp": [],
            "udp": []
        },
        {
            "action": "allow",
            "destination": "10.0.0.0/8",
            "direction": "inbound",
            "icmp": [],
            "id": "223b0b6d-2f4b-409f-b129-d038c6fa62a5",
            "ip_version": "ipv4",
            "name": "ibmflow-allow-vpc-connectivity-inbound-33aa",
            "source": "10.20.20.0/24",
            "subnets": 7,
            "tcp": [],
            "udp": []
        },
        {
            "action": "allow",
            "destination": "10.0.0.0/8",
            "direction": "inbound",
            "icmp": [],
            "id": "24e123b3-f691-49de-a28b-e97bd430f6fb",
            "ip_version": "ipv4",
            "name": "ibmflow-allow-vpc-connectivity-inbound-13be",
            "source": "10.10.30.0/24",
            "subnets": 7,
            "tcp": [],
            "udp": []
        },
        {
            "action": "allow",
            "destination": "10.0.0.0/8",
            "direction": "inbound",
            "icmp": [],
            "id": "0bf752f3-77bd-4e47-86b0-bbc687c3aaf3",
            "ip_version": "ipv4",
            "name": "ibmflow-allow-vpc-connectivity-inbound-6c66",
            "source": "10.30.10.0/24",
            "subnets": 7,
            "tcp": [],
            "udp": []
        },
        {
            "action": "allow",
            "destination": "10.0.0.0/8",
            "direction": "inbound",
            "icmp": [],
            "id": "c23eda9e-22e0-4f86-a123-495324a44bdd",
            "ip_version": "ipv4",
            "name": "ibmflow-allow-vpc-connectivity-inbound-db9f",
            "source": "10.30.20.0/24",
            "subnets": 7,
            "tcp": [],
            "udp": []
        },
        {
            "action": "allow",
            "destination": "10.0.0.0/8",
            "direction": "inbound",
            "icmp": [],
            "id": "8b118a02-146f-4045-a77c-ab5bc76c3b11",
            "ip_version": "ipv4",
            "name": "ibmflow-allow-vpc-connectivity-inbound-5cb2",
            "source": "10.20.10.0/24",
            "subnets": 7,
            "tcp": [],
            "udp": []
        },
        {
            "action": "allow",
            "destination": "10.0.0.0/8",
            "direction": "inbound",
            "icmp": [],
            "id": "71e872aa-12e1-45ac-8ba7-ba3f97363299",
            "ip_version": "ipv4",
            "name": "ibmflow-allow-vpc-connectivity-inbound-7fc9",
            "source": "10.10.10.0/24",
            "subnets": 7,
            "tcp": [],
            "udp": []
        },
        {
            "action": "allow",
            "destination": "10.10.20.0/24",
            "direction": "outbound",
            "icmp": [],
            "id": "cf33488c-81c9-4a3f-9781-bbedc6568378",
            "ip_version": "ipv4",
            "name": "ibmflow-allow-vpc-connectivity-outbound-3e78",
            "source": "10.0.0.0/8",
            "subnets": 7,
            "tcp": [],
            "udp": []
        },
        {
            "action": "allow",
            "destination": "10.20.20.0/24",
            "direction": "outbound",
            "icmp": [],
            "id": "70d703bb-4c85-4577-bb01-f0e6efe70ac9",
            "ip_version": "ipv4",
            "name": "ibmflow-allow-vpc-connectivity-outbound-33aa",
            "source": "10.0.0.0/8",
            "subnets": 7,
            "tcp": [],
            "udp": []
        },
        {
            "action": "allow",
            "destination": "10.10.30.0/24",
            "direction": "outbound",
            "icmp": [],
            "id": "9f02131b-4cd3-4ddd-8da7-b09bf5a9881a",
            "ip_version": "ipv4",
            "name": "ibmflow-allow-vpc-connectivity-outbound-13be",
            "source": "10.0.0.0/8",
            "subnets": 7,
            "tcp": [],
            "udp": []
        },
        {
            "action": "allow",
            "destination": "10.30.10.0/24",
            "direction": "outbound",
            "icmp": [],
            "id": "993265b3-7186-4d7c-82f8-c38f263edfd6",
            "ip_version": "ipv4",
            "name": "ibmflow-allow-vpc-connectivity-outbound-6c66",
            "source": "10.0.0.0/8",
            "subnets": 7,
            "tcp": [],
            "udp": []
        },
        {
            "action": "allow",
            "destination": "10.30.20.0/24",
            "direction": "outbound",
            "icmp": [],
            "id": "feddc303-8451-4503-86dd-7c5ece56d3cd",
            "ip_version": "ipv4",
            "name": "ibmflow-allow-vpc-connectivity-outbound-db9f",
            "source": "10.0.0.0/8",
            "subnets": 7,
            "tcp": [],
            "udp": []
        },
        {
            "action": "allow",
            "destination": "10.20.10.0/24",
            "direction": "outbound",
            "icmp": [],
            "id": "8ed0d99c-e430-40d8-8baf-fc88d5570b6e",
            "ip_version": "ipv4",
            "name": "ibmflow-allow-vpc-connectivity-outbound-5cb2",
            "source": "10.0.0.0/8",
            "subnets": 7,
            "tcp": [],
            "udp": []
        },
        {
            "action": "allow",
            "destination": "10.10.10.0/24",
            "direction": "outbound",
            "icmp": [],
            "id": "0909586b-8f2c-4dcc-89d3-ca6b60456c52",
            "ip_version": "ipv4",
            "name": "ibmflow-allow-vpc-connectivity-outbound-7fc9",
            "source": "10.0.0.0/8",
            "subnets": 7,
            "tcp": [],
            "udp": []
        },
        {
            "action": "deny",
            "destination": "0.0.0.0/0",
            "direction": "inbound",
            "icmp": [],
            "id": "eabf7845-906a-44c8-9481-06851dae8805",
            "ip_version": "ipv4",
            "name": "ibmflow-deny-all-inbound",
            "source": "0.0.0.0/0",
            "subnets": 7,
            "tcp": [],
            "udp": []
        },
        {
            "action": "deny",
            "destination": "0.0.0.0/0",
            "direction": "outbound",
            "icmp": [],
            "id": "63a3a3fe-d567-4ad2-b6e6-41d432db2a35",
            "ip_version": "ipv4",
            "name": "ibmflow-deny-all-outbound",
            "source": "0.0.0.0/0",
            "subnets": 7,
            "tcp": [],
            "udp": []
        }
    ]
}

By comparing both will see the duplicate ACLs are getting removed-

image

@vburckhardt Please let me know if this expected result you are looking for.

**Comment by toddgiguere** [comment]
@Ak-sky another way to present the final output would be to deploy the SLZ VPC, VSI, and maybe even the ROKS patterns using all defaults (supply only required vars like prefix/region), and after deployment take note of the actual ACL rules that were deployed. Like in the actual cloud UI or similar (ignore Terraform output or state). This "true" state represents the actual network rules and we would need to verify them to see if they are too inclusive, or we are missing something.

**Comment by Ak-sky** [comment]
1- The default deployment of `VPC pattern` results in the following ACL list for the - 

-> management VPC
- [ ] Approved
- [ ] Not Approved

image

-> workload VPC
- [ ] Approved
- [ ] Not Approved

image

---

2- The default deployment of `VSI pattern` results in the following ACL list for the - 

-> management VPC
- [ ] Approved
- [ ] Not Approved

image

-> workload VPC
- [ ] Approved
- [ ] Not Approved

image

---

3- The default deployment of `ROKS pattern` results in the following ACL list for the - 

-> management VPC
- [ ] Approved
- [ ] Not Approved

image

-> workload VPC
- [ ] Approved
- [ ] Not Approved

image

@vburckhardt, if this looking good please provide your approval on the checkboxes.
@toddgiguere thank you for this tip.

**Comment by Ak-sky** [comment]
> @Ak-sky Can you fill in the release notes section please? Ensure to include what the consumers experience will be (aka what happens to the current deployed ACLs)

@ocofaigh I have updated the release notes, please have a look and let me know If I am missing anything.

File: tests/pr_test.go [hunk 1]
```diff
@@ -29,6 +29,8 @@ const add_atracker_route = false
 var ignoreUpdates = []string{
 	"module.landing_zone.module.landing_zone.ibm_atracker_target.atracker_target[0]",
 	"module.landing_zone.ibm_atracker_target.atracker_target[0]",
+	"module.landing_zone.module.vpc[\"management\"].ibm_is_network_acl.network_acl[\"management-acl\"]",
```
Comment (toddgiguere) [inline tests/pr_test.go:32]: Why are ignoreUpdates being added to the pr_test for network acl? Was it to prevent the upgrade test from failing?

**Review by toddgiguere — CHANGES_REQUESTED** [review]

File: tests/pr_test.go (hunk 1 above)
Comment (ocofaigh) [reply tests/pr_test.go:32]: I think the answer is that we are expecting ACLs rules to be deleted from the ACL resource. But not the resource itself, so they come in as updates that we need to ignore (for this PR only)

Comment (toddgiguere) [reply tests/pr_test.go:32]: Resolving this, please ignore, was explained that this exception will be removed immediately after patch is released.

**Review by vburckhardt — APPROVED** [review]
👍

**Pull request merged**